### DISCUSSION TITLE
Be able to Set/Remove LocalAuthentication.

### DISCUSSION THREAD

**Discussion by MauriMiguez** [auth0/auth0-flutter#597] [General]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/auth0-flutter/tree/main/auth0_flutter#readme), [Examples](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/EXAMPLES.md), and [FAQ](https://github.com/auth0/auth0-flutter/blob/main/auth0_flutter/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://pub.dev/documentation/auth0_flutter/latest/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/auth0-flutter/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

In the particular App im working on, we would like to be able to toggle on or off LocalAuthentication based on user preference. Right now I don't see a way to either remove or add LocalAuthentication from an Auth0 or CredentialsManager already created instances.

To be more specific: if you already instantiated Auth0 without LocalAuthentication there is no way you can add it now.

### Describe the ideal solution

I would for the API to expose a method to be able to add/remove local authentication.

### Alternatives and current workarounds

The only work around I can think of is to make another instance of Auth0 in which CredentialsManager will be injected with the new LocalAuthentication.

### Additional context

Currently I create the Auth0 instance during the initialization of the app and after that is injected into the corresponding classes that need that instance. So is at that point that I decide to either add local biometrics or not add it. 

If user decides that now it wants to add it I would have to re-instantiate the object and inject it again which is troublesome.

**Comment by Widcket** [comment]
Hi @MauriMiguez, thanks for raising this.

Implementing this would require changes in the underlying SDKs ([Auth0.Android](https://github.com/auth0/Auth0.Android) and [Auth0.swift](https://github.com/auth0/Auth0.swift)). We don't have any plans for this feature in the short term, but I'll leave this issue open for future planning and reference.

**Comment by pmathew92** [comment]
As we don't plan to support this in our immediate roadmap, we would love to hear the community opinion and different use cases for this request, so that we can efficiently plan to add this
